src/som/compiler/MixinBuilder.java

```diff
@@ -165,16 +175,20 @@ public MixinScope getInstanceScope() {
     return instanceScope;
   }
 
-  public MixinBuilder getOuterBuilder() {
-    return outerBuilder;
+  public MixinBuilder getOuterMixinBuilder() {
+    return outerMixinBuilder;
+  }
+
+  public MethodBuilder getOuterMethodBuilder() {
+    return outerMethodBuilder;
   }
```

Comment:
Newspeak has only lexical scoping, not dynamic scoping.
Since you didn't declare any relation between `A` and `B`, the literal object does only know about `A`. But, in principle, it would close over all other outer scopes as well, if there would be any.

```
class OuterA = (|outX = 0.|) (
  class A  = ( |x = 1.| )(
    public foo = (
      | y |
      y := 11.
      ^ objL()(
          public printXY = (
            outX print. ' ' print.
            x print. ' ' print.
            y println.
      ) )
    )
  )
)

OuterA new A new foo printXY  ==>> '0 1 11\n'
```